ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -147,20 +184,45 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
                     title: workflowOperation.title.charAt(0).toUpperCase() + workflowOperation.title.slice(1),
                     iconClassName: workflowOperation.iconClassName,
                     action: () => {
-                        popup.confirm('Confirm', `Are you sure you want to ${workflowOperation.title.toLowerCase()} this workflow?`).then(yes => {
-                            if (yes) {
-                                workflowOperation
-                                    .action(workflow)
-                                    .then((wf: Workflow) => {
-                                        if (workflowOperation.title === 'DELETE') {
-                                            navigation.goto(uiUrl(`workflows/${workflow.metadata.namespace}`));
-                                        } else {
-                                            setName(wf.metadata.name);
+                        if (workflowOperation.title === 'DELETE') {
+                            popup
+                                .confirm('Confirm', () => <DeleteCheck isWfInDB={isWfInDB} isWfInCluster={isWfInCluster} />)
+                                .then(yes => {
+                                    if (yes) {
+                                        if (isWfInCluster) {
+                                            services.workflows
+                                                .delete(workflow.metadata.name, workflow.metadata.namespace)
+                                                .then(() => {
+                                                    setIsWfInCluster(false);
+                                                })
+                                                .catch(setError);
+                                        }
+                                        if (isWfInDB && (globalDeleteArchived || !isWfInCluster)) {
+                                            services.workflows
+                                                .deleteArchived(workflow.metadata.uid, workflow.metadata.namespace)
+                                                .then(() => {
+                                                    setIsWfInDB(false);
+                                                })
+                                                .catch(setError);
                                         }
-                                    })
-                                    .catch(setError);
-                            }
-                        });
+                                        navigation.goto(uiUrl(`workflows/${workflow.metadata.namespace}`));
+                                        // TODO: This is a temporary workaround so that the list of workflows
+                                        //  is correctly displayed. Workflow list page needs to be more responsive.
+                                        window.location.reload();
```

Comment:
I couldn't reproduce this issue, so I have a fix going out to remove this. It's possible this was happening due to the bug that #11711 fixed, as that would prevent Workflows from being deleted